src/octoprint/server/util/__init__.py

```diff
@@ -234,6 +234,17 @@ def get_user_for_remote_user_header(
         )
         user = octoprint.server.userManager.find_user(userid=header)
 
+    if user:
+        groupHeader = request.headers.get(
+            settings().get(["accessControl", "remoteGroupsHeader"])
+        )
+        if groupHeader:
```

Comment:
I think, just to be on the safe side, we should also check for a `trustRemoteGroups` setting here.

I could imagine existing setups out there were whatever is sending the user header is also already sending an empty group header, and that would immediately break the existing user group assignment then. Having to enable the use of sent groups would avoid this.

The risk is minimal, but it's there, and I'd rather make sure that this change doesn't have any chance to break existing workflows.